source/plugin/offline-userplayer-data.rst

```diff
@@ -0,0 +1,50 @@
+===================
+Offline Player Data
+===================
+
+It may be necessary for plugins to access player data even when the player is offline.
+You might think that ``Sponge.getServer().getPlayer()`` returning a ``Player`` can be used for this.
+But since ``Player`` objects only exist for online players, another solution must be used.
+
+Some plugins store the relevant data themselves and associate the user by using the ``GameProfileManager``.
+But writing different code for offline and online users is not necessary:
+The ``ServiceManager`` natively provides a service known as the ``UserStorageService`` which is capable of returning ``User`` instances for ``Player``\s who are currently offline.
+Since the ``Player`` interface extends ``User`` most methods you call on a ``Player`` are also available. 
+
+For example:
+
+* ``#hasPermission(String permission)`` is available from both instances.
+
+Code Example
+------------
+
+Here's an example for a utility method that can be used to get a ``User``:
+
+.. code-block:: java
+    
+    import org.spongepowered.api.Sponge;
+    import org.spongepowered.api.entity.living.player.User;
+    import org.spongepowered.api.service.user.UserStorageService;
+    
+    public Optional<User> getUser(UUID uuid) {
+        
```

Comment:
No, if anything it makes it look messy. It also doesn't match with the [Code Style](https://docs.spongepowered.org/master/en/contributing/implementation/codestyle.html).